### PR TITLE
Add cuda tags and unit test

### PR DESCRIPTION
**Pull request by zxybazh**
This commit added cuda tags of most cuda devices for target specification and completed cuda target tag attributes.

### CODE REVIEW

**Comment by zxybazh** [comment]
@junrushao1994  Please review, thanks!

File: src/target/tag.cc [hunk 1]
```diff
@@ -21,9 +21,22 @@
...
+TVM_REGISTER_TARGET_TAG(Name)             \
+    .set_config({                                       \
+        {"kind", String("cuda")},                       \
+        {"arch", String(Arch)},                         \
+        {"shared_memory_per_block", Integer(SharedMem)},\
+        {"registers_per_block", Integer(RegPerBlock)},  \
+        {"max_threads_per_block", Integer(1024)},       \
+        {"thread_warp_size", Integer(32)},              \
+    });
+
```
Comment (junrushao1994) [inline src/target/tag.cc]: Please move the macro definition to immediately before `TVM_REGISTER_CUDA_TAG `, and undef it once it is used

**Review by junrushao1994 — CHANGES_REQUESTED** [review]
Overall it looks to me. Please see my only comment below. Let's get it in quickly after fixing the linter

File: src/target/tag.cc [hunk 2]
```diff
@@ -66,12 +79,252 @@ Target TargetTag::AddTag(String name, Map<String, ObjectRef> config, bool overri
...
-/**********  Register Target tags  **********/
-
-TVM_REGISTER_TARGET_TAG("nvidia/rtx2080ti")
-    .set_config({
-        {"kind", String("cuda")},
-        {"arch", String("sm_75")},
-    });
+/**********  Register Target tags  ***********/
 
+  TVM_REGISTER_CUDA_TAG("nvidia/tesla-k80", "sm_37", 49152, 65536);
```
Comment (junrushao1994) [inline src/target/tag.cc:84]: Wait, shall we remove the indent here too?

Comment (zxybazh) [reply src/target/tag.cc]: Fixed.

File: src/target/tag.cc (hunk 1 above)
Comment (zxybazh) [reply src/target/tag.cc]: Fixed.

**Review by junrushao1994 — APPROVED** [review]
Looks good to me! Thanks Xiyou for the first contribution 🎉

**Comment by junrushao1994** [comment]
@comaniac Would you mind taking a second look?

**Comment by comaniac** [comment]
Thanks @zxybazh @junrushao1994 
It seems to me that we should be consistent to let all fields have a default value (or not), but we could do this later.

**Comment by junrushao1994** [comment]
Thanks @comaniac! For reference, we crawled the information from:

[1] https://developer.nvidia.com/cuda-gpus
[2] https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#features-and-technical-specifications

**Pull request merged**